### ISSUE TITLE
reset function in src/evaluation/eval_whitebox.py

### ISSUE THREAD

**Issue by ScottLiao920** [softsys4ai/athena#36]
Hi, is it possible to implement reset function for other transformations?

**Comment by MENG2010** [comment]
Actually, the reset function is not used anymore, it will be removed after refactored.

reset a transformation operation is not one of our future work, so, even if we keep this function, we do not have a plan to extend it in a close future. I am sorry for that. I'd encourage you to do this by yourself.

**Comment by ScottLiao920** [comment]
My concern is, currently adversarial examples are generated based on transformed images and without   a proper reset function applied before inference on them. This may lead to problems like shift_top a shift_top-ed image.

**Comment by MENG2010** [comment]
If you check Algorithm 1, none of the images (x, x', and $x^{'}_{tmp}$) are transformed. When generating perturbation based on x' (function attacker.getPerturbation($f_{target}$, x')), we first made a copy of x', then transformed and perturbed the copy. So, we only incrementally added perturbations non-transformed images. It won't impact model inference. BTW, in practice, it is not easy to implement a reverse function for some type of transformation (for some, it is impossible).

At first, we perturbed the intermediate $x'$ directly so I need the reset function for the same reason you mentioned, and it turned out that it's impossible to implement reverse functions for all transformations. Then we improved the approach to only extract and add the perturbations.

Hope this answers your question.
